src/NATS.Client.KeyValueStore/NatsKVStore.cs

```diff
@@ -190,44 +219,44 @@ public async ValueTask<NatsKVEntry<T>> GetEntryAsync<T>(string key, ulong revisi
             if (direct is { Headers: { } headers } msg)
             {
                 if (headers.Code == 404)
```

Comment:
TBH stuffing this synchronous validation that fetches a value or two, probably should get moved into static method.  (AFAIK this can be moved to static although the return may change due to the sequence parsing).

That will improve overall state machine quality and likely improve performance as a result. I peeked at the IL for something close to current main and there's lots of locals and large state machines involved. Sometimes moving certain parts of logic to a static or even separate method can help with that.

Additionally, it's probably worth at least filing an issue to see whether splitting these 'if/else' branches into sub-calls gives a performance or memory boost due to likelyhood State machine will allocate for things only in one path...

Reply:
moved to #691